tslearn/nonmyopic.py

```diff
@@ -0,0 +1,450 @@
+from tslearn.utils import to_time_series_dataset
+from tslearn.clustering import TimeSeriesKMeans, silhouette_score
+import matplotlib.pyplot as plt
+import numpy as np
+import math
+from sklearn.metrics import roc_auc_score
+import argparse
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--cost_time_parameter', type=float, default=0.01)
+args = parser.parse_args()
+
+
+class NonMyopicEarlyClassification(BaseEstimator, ClassifierMixin):
+    """Early Classification modelling for time series using the model presented in [1]
+
+    Parameters
+    ------------------
+    number_cluster : int
+        Number of clusters to form.
+
+    solver :
+        The method of gradient descent used during training
+
+    hidden_layer_sizes : tuple of int
+        Size of the hidden layers
+
+    random_state : int
+        Seed of the MLP
+
+    maximum_iteration : int
+        Maximum number of iterations performed by the MLP
+
+    minimum_time_stamp : int
+        Earliest time at which a classification can be performed on a time series
+
+    lamb : float
+        Value of the hyper parameter lambda
+
+    cost_time_parameter : float
+        Parameter of the cost function of time
+
+    Attributes
+    --------------------
+    silhouette_ : float
+        The silhouette score from the clustering
+
+    classifier_ : list
+        A list containing all the classifiers trained for the model
+
+    clusters_ : dictionary
+        Contains the times series by clusters
+
+    pyhatyck_ : dictionary
+        Contains the probabilities of being classified as class y_hat given class y and cluster ck
+
+    indice_ck_ : list
+        Contains for each clusters the indexes of the time series in the dataset
+
+    number_classes_ : int
+        The number of classes in the series
+
+    len_X_ : int
+        The length of the time series
+
+    Examples
+    --------------------
+    >>>
+
+    References
+    --------------------
+    [1] A. Dachraoui, A. Bondu & A. Cornuéjols. Early classification of time series as a non myopic sequential decision
+    making problem. 2015 Conference paper
+
+
+    """
+
+    def __init__(self, number_cluster, solver, hidden_layer_sizes, random_state, maximum_iteration, minimum_time_stamp,
+                 lamb, cost_time_parameter):
+        super(NonMyopicEarlyClassification, self).__init__()
+        self.solver = solver
+        self.hidden_layer_sizes = hidden_layer_sizes
+        self.random_state = random_state
+        self.maximum_iteration = maximum_iteration
+        self.number_cluster = number_cluster
+        self.minimum_time_stamp = minimum_time_stamp
+        self.lamb = lamb
+        self.cost_time_parameter = cost_time_parameter
+
+    def clustering(self, X_train):
+        """
+        This function compute and fit clusters for a training dataset of time series using K-means and euclidean
+        distances
+        Parameters
+        -----------------
```

Comment:
The length of the underline should be exactly the same as the title above, otherwise you would get sphinx warnings. Here and everywhere else.